spec/migrations/20181016140921_migrate_orch_stacks_to_have_ownership_concept_spec.rb

```diff
@@ -0,0 +1,63 @@
+require_migration
+
+describe MigrateOrchStacksToHaveOwnershipConcept do
+  let(:orchestration_stack) { migration_stub(:OrchestrationStack) }
+  let(:ems) { migration_stub(:ExtManagementSystem) }
+  let(:service) { migration_stub(:Service) }
+  let(:tenant) { migration_stub(:Tenant).create }
+  let(:group) { migration_stub(:MiqGroup).create(:tenant => tenant) }
+  let!(:user) { migration_stub(:User).create(:userid => "admin", :miq_groups => [group], :current_group => group) }
+
+  migration_context :up do
+    it "does nothing if the ems and service do not exist" do
+      stack = orchestration_stack.create(:ext_management_system => nil)
+      expect(orchestration_stack.count).to eq(1)
+
+      migrate
+
+      expect(stack.tenant_id).to be(nil)
+      expect(stack.evm_owner_id).to be(nil)
+      expect(stack.miq_group_id).to be(nil)
+    end
+
+    it "does something if ems exists and service doesn't" do
+      ext_ms = ems.create(:tenant_id => tenant.id)
+      stack = orchestration_stack.create(:ext_management_system => ext_ms)
+      expect(orchestration_stack.count).to eq(1)
+
+      migrate
+      stack.reload
+
+      expect(stack.evm_owner_id).to eq(ext_ms.tenant_identity.id)
+      expect(stack.tenant_id).to eq(ext_ms.tenant_id)
+      expect(stack.miq_group_id).to eq(ext_ms.tenant_identity.current_group.id)
+    end
+
+    it "does something if the service exists but no ems" do
```

Comment:
"sets owner, tenant, and group from the service if the service exists and ems doesn't"